packages/system/src/Item/Components/ItemSearchInput/ItemSearchInput.tsx

```diff
@@ -28,41 +28,61 @@ const renderOption = (
 );
 
 interface ItemSearchInputProps {
-  onChange: (item: Item) => void;
+  onChange: (item: Item | null) => void;
   currentItem?: Item;
-  currentItemId?: string;
+  currentItemCode?: string;
 }
 
 export const ItemSearchInput: FC<ItemSearchInputProps> = ({
   onChange,
   currentItem,
-  currentItemId,
+  currentItemCode,
 }) => {
-  const { data, isLoading } = useItems();
+  const [filter, setFilter] = useState({
+    searchTerm: currentItem?.code ?? currentItemCode ?? '',
+    field: 'code',
+  });
+
+  const { data, isLoading, onFilterByName, onFilterByCode } = useItemsList({
+    initialSortBy: { key: 'name' },
+    initialFilterBy: currentItem?.code
+      ? { code: { equalTo: currentItem?.code } }
+      : undefined,
+  });
   const t = useTranslation();
 
+  useEffect(() => {
+    setFilter({
+      searchTerm: currentItem?.code ?? currentItemCode ?? '',
+      field: 'code',
+    });
+  }, [currentItem, currentItemCode]);
+
+  useEffect(() => {
+    if (filter.field === 'name') return onFilterByName(filter.searchTerm);
+    if (filter.field === 'code') return onFilterByCode(filter.searchTerm);
+  }, [filter]);
+
   const value =
     currentItem ??
-    (currentItemId
-      ? data?.nodes?.find(i => i.id === currentItemId)
-      : undefined);
+    (currentItemCode
+      ? data?.nodes?.find(i => i.code === currentItemCode) || null
+      : null);
```

Comment:
oh. yes. that last comment - the item doesn't need to be in the list really. Though the autocomplete should look like it has the selected item, and the autocomplete should still function correctly; so it would make sense to do the dance that you have, and make it all work.